subsys/nrf_security/src/drivers/cracen/cracenpsa/src/sign.c

```diff
@@ -306,14 +344,33 @@ static psa_status_t cracen_signature_ecc_sign(int message, const psa_key_attribu
 	sign.r = (char *)signature;
 	sign.s = (char *)signature + *signature_length / 2;
 
-	if (message) {
+	/* ED25519PH requires prehashing and supports sign and verify message
+	 * the message is therefore hashed here before si_sig_verify is called
+	 */
+	if (alg == PSA_ALG_ED25519PH && message) {
+		uint8_t status;
+		uint8_t hash[64];
+		size_t output_len;
+
+		status = psa_hash_compute(PSA_ALG_SHA_512,
+				input,
+				input_length,
+				hash,
+				64,
+				&output_len);
+
 		si_sig_create_sign(&t, &privkey, &sign);
-		si_task_consume(&t, (char *)input, input_length);
+		si_task_consume(&t, (char *)hash, 64);
 	} else {
-		si_sig_create_sign_digest(&t, &privkey, &sign);
-		si_task_consume(&t, (char *)input, sx_hash_get_alg_digestsz(privkey.hashalg));
+		if (message) {
+			si_sig_create_sign(&t, &privkey, &sign);
+			si_task_consume(&t, (char *)input, input_length);
+		} else {
+			si_sig_create_sign_digest(&t, &privkey, &sign);
+			si_task_consume(&t, (char *)input,
+					sx_hash_get_alg_digestsz(privkey.hashalg));
+		}
```

Comment:
I don't know. if (message) { ..} else is the convention used throughout the file. See :153, :176,  :218. I can change it, but then it should probably be updated throughout the entire file

Reply:
In the cases you mention it's not possible because the function signatures are different (one takes an additional argument). Do as you want, I'm just a bit concerned with the amount of (unnecessary) lines of code and the duplication.

Reply:
You are right, in the other cases it can't be done as easily. Changing it